File: plugins/piranhas_2019/server/sc/plugin2019/GamePlugin.java
package sc.plugin2019;

import sc.api.plugins.IGameInstance;
import sc.api.plugins.IGamePlugin;
import sc.api.plugins.host.IGamePluginHost;
import sc.plugin2019.util.Configuration;
import sc.plugins.PluginDescriptor;
import sc.shared.ScoreAggregation;
import sc.shared.ScoreDefinition;
import sc.shared.ScoreFragment;

/**
 * Abstrakte Beschreibung des Spieles Prianahs für die Software-Challenge 2019
 */
@PluginDescriptor(name = "Piranhas", uuid = GamePlugin.PLUGIN_UUID, author = GamePlugin.PLUGIN_AUTHOR)
public class GamePlugin implements IGamePlugin {

  public static final String PLUGIN_AUTHOR = "";
  public static final String PLUGIN_UUID = "swc_2019_piranhas";

  public static final ScoreDefinition SCORE_DEFINITION;

  static {
    SCORE_DEFINITION = new ScoreDefinition();
    SCORE_DEFINITION.add("Gewinner");
    // NOTE: Always write the XML representation of unicode characters, not the character directly, as it confuses the
    // parsers which consume the server messages!
    SCORE_DEFINITION.add(new ScoreFragment("\u2205 Schwarm", ScoreAggregation.AVERAGE));
  }

  @Override
  public IGameInstance createGame() {
    return new Game(PLUGIN_UUID);
  }

  @Override
  public void initialize(IGamePluginHost host) {
    host.registerProtocolClasses(Configuration.getClassesToRegister());
  }

  @Override
  public void unload() {
    // TODO Plugin entladen
  }

  @Override
  public ScoreDefinition getScoreDefinition() {
    return SCORE_DEFINITION;
  }

}

File: plugin/src/main/sc/plugin2022/GamePlugin.kt
package sc.plugin2022

import sc.api.plugins.IGameInstance
import sc.api.plugins.IGamePlugin
import sc.api.plugins.IGameState
import sc.plugin2022.util.Constants
import sc.plugins.PluginDescriptor
import sc.shared.ScoreAggregation
import sc.shared.ScoreDefinition
import sc.shared.ScoreFragment

@PluginDescriptor(name = "Blokus", uuid = GamePlugin.PLUGIN_ID)
class GamePlugin: IGamePlugin {
    companion object {
        const val PLUGIN_ID = "swc_2021_blokus"
        val scoreDefinition: ScoreDefinition =
                ScoreDefinition(arrayOf(
                        ScoreFragment("Gewinner"),
                        ScoreFragment("Bernsteine", ScoreAggregation.AVERAGE),
                        ScoreFragment("Figur vorne", ScoreAggregation.AVERAGE)
                ))
    }
    
    override val id = PLUGIN_ID
    
    override val scoreDefinition =
            Companion.scoreDefinition
    
    override val gameTimeout: Int =
            Constants.GAME_TIMEOUT
    
    override fun createGame(): IGameInstance =
            Game()
    
    override fun createGameFromState(state: IGameState): IGameInstance =
            Game(state as GameState)
    
}
